content/guides/java/inject-extract.md

```diff
@@ -2,8 +2,142 @@
 title: "Java: Inject and Extract"
 ---
 
-* When do you use Inject and Extract?
-* Carriers and Format Options
-* Injecting and Extracting HTTP
-* Injecting and Extracting TextMap
-* Injecting and Extracting Binary
+## Introduction
+
+In order to trace across process boundaries and RPC calls in distributed systems, span context needs to propagated over the wire. The Java Opentracing API provides two functions in the Tracer interface to do just that, `inject(SpanContext, format, carrier)` and `extract(format, carrier)`.
+
+## Format Options and Carriers
+The **format** parameter refers to one of the three standard encodings the OpenTracing API defines:
+
+1. `TEXT_MAP` where span context is encoded as a collection of string key-value pairs,
```

Comment:
Not sure how others feel about it, but I'd prefer to have `SpanContext` instead of "span context"